plugins/kubernetes/test/models/kubernetes/api/horizontal_pod_autoscaler_test.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+require_relative '../../../test_helper'
+
+SingleCov.covered!
+
+describe Kubernetes::Api::HorizontalPodAutoscaler do
+  let(:start_time) { "2017-03-31T22:56:20Z" }
+  let(:event) { {metadata: {creationTimestamp: start_time}, type: 'Normal'} }
+  let(:events) { [event] }
+  let(:hpa) { Kubernetes::Api::HorizontalPodAutoscaler.new }
+
+  describe "HorizontalPodAutoscaler with failures we don't care about" do
+    before do
+      event.merge!(kind: 'HorizontalPodAutoscaler', type: 'Warning')
+    end
+
```

Comment:
start with a test that modifies nothing and asserts things are ok `it "does not ignore bad events"` or so, then each subsequent test changes 1 aspect and tests that the outcome change